engine/src/flutter/testing/skia_gold_client/lib/skia_gold_client.dart

```diff
@@ -88,6 +88,12 @@ interface class SkiaGoldClient {
       path.join(_engineRoot.flutterDir.path, '.engine-release.version'),
     );
 
+    if (!_isMainBranch) {
+      _stderr.writeln(
```

Comment:
nit: I seems like skipping the check for non-main branches is our system WAI, and so I wouldn't expect to see output into stderr.

Reply:
Doesn't writing to stderr automatically fail some tests?

Reply:
I hope not! I can make this a verbose only check if we're concerned but let's give it a shot first.

Reply:
My comment was just a nit. If you thought about it and stderr still looks like the better choice, then SGTM.